test-harness/tests/test_master.py

```diff
@@ -64,21 +64,38 @@ def test_if_upstream_headers_are_correct(self,
 
 
 class TestSystemLoggingAgentEndpoint():
+    @pytest.mark.parametrize("prefix", [("/mesos")])
+    @pytest.mark.parametrize("agent,endpoint", [
+        ("de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1", 'http://127.0.0.2:15001')])
+    def test_if_request_is_sent_to_correct_mesos_upstream(self,
+                                                          master_ar_process,
+                                                          valid_user_header,
+                                                          agent,
+                                                          endpoint,
+                                                          prefix):
+
+        # FIXME - these are very simple tests for now, need to think how to test
+        # streaming api better. ATM we only test if HTTP is set to 1.1 for streaming
+        # stuff.
+        generic_correct_upstream_dest_test(master_ar_process,
+                                           valid_user_header,
+                                           '/system/v1/agent/{}{}/foo/bar'.format(agent, prefix),
+                                           endpoint)
+
     @pytest.mark.parametrize("prefix", [
-        (""),
         ("/logs/v1"),
         ("/metrics/v0"),
     ])
     @pytest.mark.parametrize("agent,endpoint", [
         ("de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1", 'http://127.0.0.2:61001'),
         ("de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S0", 'http://127.0.0.3:61001'),
     ])
-    def test_if_request_is_sent_to_correct_upstream(self,
-                                                    master_ar_process,
-                                                    valid_user_header,
-                                                    agent,
-                                                    endpoint,
-                                                    prefix):
+    def test_if_request_is_sent_to_correct_logs_or_metrics_upstream(self,
```

Comment:
the class in which these tests reside in already provides you this context. If you want to filter this specific test, you can use multiple `-k` params and first filter on class and then on the test name.